Recon/src/Recon/SequentialTrigger.cxx

```diff
@@ -0,0 +1,71 @@
+#include "Recon/SequentialTrigger.h"
+#include "Recon/Event/TriggerResult.h"
+
+namespace recon {
+
+/**
+ *Instantiates the variables used in this processor. We take trigger_list_ and tigger_passNames_ to obtain trigger collections, a doOR_ and doAND_ option to check if one of or all events pass, and doVAL_ to output a validation collection
+ * */
+
+void SequentialTrigger::configure(framework::config::Parameters &ps) {
+  trigger_list_ = ps.getParameter<std::vector<std::string>>("trigger_list");
+  trigger_passNames_ = ps.getParameter<std::vector<std::string>>("trigger_passNames");
+  doAND_ = ps.getParameter<bool>("doAND");
+  doOR_ = ps.getParameter<bool>("doOR");
+  doVAL_ = ps.getParameter<bool>("doVAL");
+  //Returns an error if some bad combination of doOR and doAND is enabled.
+  if (doAND_ == doOR_){
+    ldmx_log(fatal) << "Either tried to do both or neither of  doAND and doOR. Exiting.";
+    throw std::invalid_argument("Either tried to do both or neither of doAND and doOR. Exiting.");
+    return;
+  }
+  return;
+}
+
+/**
+ *
+ *This producer takes in a list of triggers and runs an OR or AND skimmer depending on the config file. It will change the keep
+ tag and produce nothing. This is unless the validation command is set to true, in which case it produces a boolean collection in an output file
+ with the keeping flag.
+ *
+ * */
+
+void SequentialTrigger::produce(framework::Event& event) {
+  bool hasPassed = false;
+  bool keepTrack = false;
+  
+  for (int i = 0; i<trigger_list_.size(); i++){
+    //Returns an error is a trigger collection DNE
+    try{
+        auto trigResult{event.getObject<ldmx::TriggerResult>(trigger_list_[i],trigger_passNames_[i])};
+    
+    //Returns true should any trigger pass and doOR_ enabled, and sets keepTrack to true and breaks if doAND_ and any fail
+        if(trigResult.passed()){
+            if(doOR_){hasPassed=true;break;}
+        }else{
+            if(doAND_){keepTrack=true;break;}
```

Comment:
why aren't you instead setting hasPassed=false here? is it because you initialize to false in both cases?

my suggestion would be to initialize to either doAND or !doOR (which should be equivalent by construction, so you choice). that way, if you run in AND mode, it is true until the first instance of a failed trigger decision. that setup warrants a descriptive comment though because just because it's simple doesn't mean it's something you grasp at first glance. i only advocate it because i think the extra keepTrack variable is also a little convoluted.

Reply:
Great idea. Done